oauth_provider/oauth2/validator.py

```diff
@@ -0,0 +1,253 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 SYLEAM
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+from datetime import datetime, timedelta
+from openerp import http
+from openerp import fields
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from oauthlib.oauth2 import RequestValidator
+except ImportError:
+    _logger.debug('Cannot `import oauthlib`.')
+
+
+class OdooValidator(RequestValidator):
+    """
+    OAuth2 validator to be used in Odoo
+    """
+    def authenticate_client(self, request, *args, **kwargs):
+        """
+        Authenticate the client
+        """
+        raise NotImplementedError
+        # return False
```

Comment:
I leaved this line here to remind the return value's type, but I may add this in the docstring.